src/EnergyPlus/HVACFourPipeBeam.cc

```diff
@@ -0,0 +1,1302 @@
+// C++ Headers
+#include <cmath>
+
+// ObjexxFCL Headers
+#include <ObjexxFCL/Array.functions.hh>
+#include <ObjexxFCL/Fmath.hh>
+
+// EnergyPlus Headers
+#include <AirTerminalUnit.hh>
+#include <HVACFourPipeBeam.hh>
+#include <BranchNodeConnections.hh>
+#include <DataContaminantBalance.hh>
+#include <DataDefineEquip.hh>
+#include <DataEnvironment.hh>
+#include <DataHVACGlobals.hh>
+#include <DataIPShortCuts.hh>
+#include <DataLoopNode.hh>
+#include <DataPlant.hh>
+#include <DataPrecisionGlobals.hh>
+#include <DataSizing.hh>
+#include <DataZoneEnergyDemands.hh>
+#include <DataZoneEquipment.hh>
+#include <FluidProperties.hh>
+#include <General.hh>
+#include <GeneralRoutines.hh>
+#include <InputProcessor.hh>
+#include <NodeInputManager.hh>
+#include <OutputProcessor.hh>
+#include <PlantUtilities.hh>
+#include <Psychrometrics.hh>
+#include <ReportSizingManager.hh>
+#include <ScheduleManager.hh>
+#include <UtilityRoutines.hh>
+#include <CurveManager.hh>
+
+
+namespace EnergyPlus {
+
+namespace FourPipeBeam {
+
+	Array1D< std::shared_ptr< HVACFourPipeBeam > > FourPipeBeams; // dimension to number of machines
+
+	// Needed for unit tests, should not be normally called.
+	void
+	HVACFourPipeBeam::clear_state()
+	{
+		FourPipeBeams.deallocate();
```

Comment:
@EnergyArchmage This is why we shouldn't have a clear_state() in classes...

Are all objects calling clear_state() or just one? What happens if you call clear_state on one object and then go to a different object and do something else? Really this clear_state() should be a function within the namespace like the other clear_states, not within the class.